framework/CodeInterfaces/RELAP5/RELAPparser.py

```diff
@@ -19,35 +19,296 @@
+
+  def getTripsMinorEditsAndControlVars(self):
+    """
+      Method to store the Trips, Minor edits and Control Variables in RELAP5
+      @ In, None
+      @ Out, None
+    """
+    for deckNum in self.deckLines.keys():
+      self.lastTripLine[deckNum]     = -1
+      self.lastCntrLine[deckNum]     = -1
+      self.lastMinorEditLine[deckNum]= -1
+      for lineNum, line in enumerate(self.deckLines[deckNum]):
+        splitted = _splitRecordAndRemoveComments(line)
+        if len(splitted) > 0 and splitted[0].strip().isdigit():
+          isMinor = self.storeMinorEdit(deckNum,splitted)
+          isTrip  = self.storeTrip(deckNum,splitted)
+          isCntr  = self.storeControlVars(deckNum,splitted)
+          if isMinor:
+            self.lastMinorEditLine[deckNum] = lineNum+1
+          if isTrip :
+            self.lastTripLine[deckNum]      = lineNum+1
+          if isCntr :
+            self.lastCntrLine[deckNum]      = lineNum+1
+
+  def getMinorEditsInfo(self):
+    """
+    """
+    for deckNum in self.deckLines.keys():
+      for lineNum, line in enumerate(self.deckLines[deckNum]):
+        splitted = _splitRecordAndRemoveComments(line)
+        if len(splitted) > 0 and splitted[0].strip().isdigit():
+          isMinor = self.storeMinorEdit(deckNum, splitted)
+          if isMinor: self.lastMinorEditLine[deckNum] = lineNum+1
+
+  def storeControlVars(self,deckNum, splitted):
+    """
+    """
+    isCntr = False
+    if int(splitted[0]) == 20500000:
+      self.controlVarType[deckNum] = 1 if splitted[1].strip() == '999' else 2
+      return isCntr
+    if deckNum not in self.inputControlVars.keys():
+      self.inputControlVars[deckNum] = {}
+      self.controlVarType[deckNum] = 1
+    if 20500010 <= int(splitted[0]) <= 20599990:
+      isCntr = True
+      if splitted[0].endswith('0'):
+        controlVar = splitted[0][3:6] if self.controlVarType[deckNum] == 1 else splitted[0][3:7]
+        if controlVar not in self.inputControlVars[deckNum].keys():
+          self.inputControlVars[deckNum][controlVar] = {'alphaNumId': splitted[1], 'cardId':splitted[0]}
+    return isCntr
+
+  def storeMinorEdit(self,deckNum, splitted):
+    """
+    """
+    isMinor = False
+    if deckNum not in self.inputMinorEdits.keys():
+      self.inputMinorEdits[deckNum] = {}
+    if 301 <= int(splitted[0]) <= 399:
+
+      if not len(splitted) == 3:
+        raise IOError(self.printTag+ "ERROR: in RELAP5 input file the number of words in minor edits section needs to be 2. Edit= "+splitted[0])
+      self.inputMinorEdits[deckNum][splitted[0].strip()], isMinor = {'variable':splitted[1].strip(),'component':splitted[2].strip()}, True
+    return isMinor
+
+  def storeTrip(self,deckNum, splitted):
+    """
+    """
+    isTrip = False
+    if deckNum not in self.inputTrips.keys():
+      self.inputTrips[deckNum] = {'variableTrips':{},'logicalTrips':{}}
+    if (401 <= int(splitted[0]) <= 599) or (20600010 <= int(splitted[0]) <= 20610000):
```

Comment:
should 20600010 be a variable in the code interface?

Reply:
nono this is a variable hardcoded in relap (it just represents the boundaries of the trip variables (in relap))